virttest/qemu_devices/qcontainer.py

```diff
@@ -2165,7 +2176,10 @@ def images_define_by_params(self, name, image_params, media=None,
             self.temporary_image_snapshots.add(image_filename)
             image_encryption = storage.ImageEncryption.encryption_define_by_params(
                     sn, sn_params)
+            image_access = storage.ImageAccessInfo.access_info_define_by_params(
+                    sn, sn_params)
             imgfmt = 'qcow2'
+            name = sn  # now image is sn instead of image1
```

Comment:
I don't suggest to replace the `name` by `sn`. This just creates an offline snapshot to blockdev mode to surpport multiple vms which the cases boot them use the "snapshot = on" on drive mode.And the argument `name` maybe be used in the case level.

Reply:
Hi Yongxue, 
given this situation, image1 is an iscsi image which needs secret object to access, sn is the local image, when forming the qemu-kvm command line, we need to associate the top image with the secret object if needed(pls refer to line 1713), here we define sn image with name=image1, then line 1713 will associate image1's secret object with sn image, which leads to unexpected error.

But your concern matters, two ways to solve this:
1. As only rh_kernel_update is affected for network storage backend, I can change rh_kernel_update that network storage is not supported.
2. set name=sn only when network storage backend is used.

What's your opinion, please?